module-services/service-cellular/src/SimCard.cpp

```diff
@@ -146,6 +146,7 @@ namespace cellular
             case at::SimState::SimPin:
                 [[fallthrough]];
             case at::SimState::SimPin2: {
+                Store::GSM::get()->sim = Store::GSM::SIM::SIM_NEED_PIN;
```

Comment:
By the way: do we ever handle/support the PIN2/PUK2?

Reply:
It seems to me that in cellular the status will be handled correctly, while in the GUI the same screen will be displayed as in the case of PIN and PUK

Reply:
Well, then the user surely won't guess that they're supposed to enter the PIN2 instead of just the usual PIN... But maybe the cause of this is outside of this PR